Kentico/delivery-sdk-net
# fixes #146

### Motivation

Which issue does this fix? Fixes #146 

* Non-generic `IInlineContentItemsResolver` was removed as it was an unusable interface (breaking change) that might eventually cause confusion in combination with `ITypelessInlineContentItemsResolver`. 
* An independent `ITypelessInlineContentItemsResolver` interface was introduced to wrap and hide `IInlineContentItemsResolver<TContentItem>` resolvers and provide them in a non-generic way to the `InlineContentItemsProcessor` that later puts inline content items through the resolver in order to obtain their string representation – I've chosen this way to mitigate a breaking change in `IInlineContentItemsResolver<>` (aka all existing resolvers).
* New extension methods for IServiceCollection were introduced to allow registration of custom inline content items resolvers (docs needs to be eventually updated).
* I have added some tests for the `IInlineContentItemsResolver<>`s to the DI framework integration tests so we can be sure that the approach would work.

### Checklist


### How to test

Follow the repro steps in #146

Note: there's another breaking change - Petr removed some methods from one interface. The methods didn't really make sense there (see https://github.com/Kentico/delivery-sdk-net/pull/147).
